### PR TITLE
Fixed TestDetailsDialog

### PR DESCRIPTION
**Pull request by Shirkanesi**
Should fix partly #193: https://github.com/kit-sdq/programming-lecture-eclipse-artemis/issues/193#issuecomment-1106109726 (worked in a test-VM; not tested on bare-metal Linux)

EDIT: I will remove the link to the issue. Otherwise, it will be closed during the merge.

### CODE REVIEW

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java [hunk 1]
```diff
@@ -21,49 +21,42 @@ class TestDetailsDialog extends Dialog {
 	private String testDetails;
 
 	public TestDetailsDialog(Shell parent, String testName, String testDetails) {
-		super(parent, SWT.DIALOG_TRIM | SWT.RESIZE | SWT.ON_TOP);
+		// super(parent, SWT.DIALOG_TRIM | SWT.RESIZE | SWT.ON_TOP);
+		super(parent);
```
Comment (dfuchss) [inline bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java:25]: At least the ON_TOP is helpful from my point of view. Can you re-add it ?

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java [hunk 2]
```diff
@@ -21,49 +21,42 @@ class TestDetailsDialog extends Dialog {
 	private String testDetails;
 
 	public TestDetailsDialog(Shell parent, String testName, String testDetails) {
-		super(parent, SWT.DIALOG_TRIM | SWT.RESIZE | SWT.ON_TOP);
```
Comment (dfuchss) [inline bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java:24]: Is it resizable by default?

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java [hunk 3]
```diff
@@ -21,49 +21,42 @@ class TestDetailsDialog extends Dialog {
...
 
-	public void open() {
-		Shell shell = new Shell(this.getParent(), this.getStyle());
-		shell.setText(this.testName);
-		shell.setSize(WIDTH, HEIGHT);
+	@Override
+	protected void configureShell(Shell newShell) {
+		super.configureShell(newShell);
+		newShell.setText(this.testName);
+		newShell.setSize(WIDTH, HEIGHT);
```
Comment (dfuchss) [inline bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java:35]: The window is not modal now?

**Review by dfuchss — CHANGES_REQUESTED** [review]
Just some questions are open for me.

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java (hunk 2 above)
Comment (Shirkanesi) [reply bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java:24]: No, I'll re-add it.

**Comment by Shirkanesi** [comment]
To address ON_TOP and modal:
I personally think forcing the window on top is a bad idea. Especially when searching a problem with protocol-test it would be way more convenient to be able to look at the code and the error message at the same time. 
However if you insist on modal and ON_TOP we can also go this route and maybe loose parts of the usefulnes of the window.

**Comment by dfuchss** [comment]
> To address ON_TOP and modal:
> I personally think forcing the window on top is a bad idea. Especially when searching a problem with protocol-test it would be way more convenient to be able to look at the code and the error message at the same time.
> However if you insist on modal and ON_TOP we can also go this route and maybe loose parts of the usefulnes of the window.

1. regarding modal: I do not want to to have the Dialog Modal (I think that was the case why I added a new shell to it (but I'm not sure)). Therefore, I just want to make sure that it's not modal now :)
2. The idea for ON_TOP was that if you are using only one monitor and want to open several classes after opening the detail view, ON_TOP makes it easier to find the detail view again. But that's discussable from my point of view. What's your opinion @majuwa

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java (hunk 3 above)
Comment (dfuchss) [reply bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/ui/TestDetailsDialog.java:35]: To be precise. It shall not be modal.

**Comment by majuwa** [comment]
Hmm, I've just tried it out and would probably prefer the ON_TOP. At least at my Linux version (Ubuntu 22.04 with Gnome) the dialog is not shown in task-switcher so it might get lost. However, this can be only a Linux problem. If for others, it is better to not use this option, it would be fine with me.

**Comment by Shirkanesi** [comment]
I agree on majuwa. I've made the window non-modal but on top. Hope everyone is happy now :)

**Review by dfuchss — APPROVED** [review]

**Pull request merged**